File: frontend/src/core/models/booking.ts
import type {  BaseType } from "./base";

export type BookingViewType = BaseType & {
    date: string,
    master: string,
    style_type: string,
    time: string,
    phone: string
}

export type MasterBookingType = Omit<BookingViewType, "created_at" | "updated_at"> & {
    username: string,
    phone: string,
    description: string
} 

export type CreateBookingType = {
    user_id: number,
    date: string,
    time: string,
    master_style_type_id: number,
    description?: string
}

export type BookingByIdType = BaseType & CreateBookingType
export type UpdateBookingType = BookingByIdType

File: backend/internal/domain/booking.go
package domain

type Booking struct {
	Base
	UserId            int    `json:"user_id"`
	MasterStyleTypeId int    `json:"master_style_type_id"`
	Date              string `json:"date"`
	Time              string `json:"time"`
	Description       string `json:"description"`
}

type BookingResponse struct {
	Base
	Master    string `json:"master" gorm:"column:master"`
	StyleType string `json:"style_type"`
	Date      string `json:"date"`
	Time      string `json:"time"`
	Phone     string `json:"phone"`
}

type CreateBookingPayload struct {
	UserId            int    `json:"user_id"`
	MasterStyleTypeId int    `json:"master_style_type_id"`
	Date              string `json:"date"`
	Time              string `json:"time"`
	Description       string `json:"description"`
}

type CreateBookingRes = SuccessResWithData[Booking]
type BookingListRes = SuccessResWithMeta[BookingResponse]
type BookingByIdRes = SuccessResWithData[Booking]
